src/zenml/config/step_configurations.py

```diff
@@ -133,6 +133,8 @@ class StepSpec(StrictBaseModel):
     source: str
     upstream_steps: List[str]
     inputs: Dict[str, InputSpec] = {}
+    # The default value is to ensure compatibility with specs of version <0.2
+    pipeline_parameter_name: str = ""
```

Comment:
The `BaseStep` attribute called `pipeline_parameter_name` has actually been the first instance of these stupid name distinctions, but I dislike both `name`/`entrypoint_name` and `name`/`pipeline_parameter_name`.
- `entrypoint_name` is simply wrong, it could be the entrypoint name, or the class name, or a manually configured name

I would suggest we don't change anything here as hopefully these duplicate names will be a thing of the past once we implement the simplified pipeline interface next sprint (in which there will only be the "step name")
```python
with pipeline(...) as p:
  step_class()
```